### PR TITLE
Support for wrk2

### LINKED ISSUE #126: Add wrk2 support for HTTP Load Test
**Feature Epic Description**
Extend Meshery to bring in the capability to work with multiple load generators as built-in libraries through an interface and load generator-specific implementation. 
Through Meshery UI, be able to choose and configure the two supported load generators: _Fortio_ and _wrk2_ for HTTP load testing.

This feature epic is comprised of these enhancement issues:
- #209 M0: Extend Load Generation Build and Packaging
- #210 M1: Extend Load Generation w/wrk2
- #211 M2: General Interface for Load Generators
- #212 M3: Incorporate wrk2 as a Load Generator
- #213 M4: Expose Load Generator settings in UI 

Resources:
https://github.com/wg/wrk
https://github.com/giltene/wrk2

### LINKED ISSUE #212: [Performance] M3: Incorporate wkr2 as a Load Generator
**Description**
See #126 for full description of Load Generator Integration Epic.

**Milestone 3: Incorporate wrk2 as a Load Generator**

- [ ] Create an implementation of the “LoadGeneratorInterface” for wrk2, “WRKLoadGenerator”
  - [ ] Which will invoke the wrk2 with the right parameters
  - [x] Extract and parse the json results and populate the “*periodic.RunnerResults” struct
  - [ ] Handle failure cases as well
  - [x] Given that wrk2 does not return the data points for the test
    - [x] We need to either extend wrk2 to be able to do that (OR)
    - [x] Make changes to the UI to behave differently for wrk2 tests
- [x] Update the load-test handler to include the load generator name as part of the request for initiating load test

**Additional context**
See [Load Generator Integration design doc](https://docs.google.com/document/d/1jZMOih3Qn1ixrq1Ve6fkdZZ4TuKNoF_qfClDZTnLNcw/edit#).

### LINKED ISSUE #477: goreleaser overwriting release description field
**Description**
"Meshery " goes missing in the release description field after goreleaser runs.

**Expected Behavior**
Each release should be named "Meshery vx.x.x".

### LINKED ISSUE #485: UI: NSM adapter chip missing NSM icon
**Expected Behavior**
The NSM icon should show on the NSM adapter chip. This issue is first seen in Meshery v0.3.0.

**Screenshots**
Screen Shot 2019-11-10 at 5 18 29 PM

### PR DESCRIPTION
**Pull request by girishranganathan**
- WRK2 integration through gowrk2
- updated docker image base
- fixed a few lint errors
- fixes #485
- fixes #477
- fixes #212
- fixes #126 
- partially addresses #213

### CODE REVIEW

**Comment by girishranganathan** [comment]
fixes #485

**Comment by girishranganathan** [comment]
fixes #477

**Comment by girishranganathan** [comment]
fixes #212

**Comment by girishranganathan** [comment]
partially fixes #213

**Comment by leecalcote** [comment]
> partially fixes #213

@girishranganathan if this is only a partial fix, will you use a verb other than "fixes" so as not to have #213 close? And update the checkboxes in #213 with what's completed?

File: .goreleaser.yml [hunk 1]
```diff
@@ -146,3 +146,31 @@ changelog:
...
+
+  # If set to auto, will mark the release as not ready for production
+  # in case there is an indicator for this in the tag e.g. v1.0.0-rc1
+  # If set to true, will mark the release as not ready for production.
+  # Default is false.
+  # prerelease: auto
+
+  # You can change the name of the GitHub release.
+  # Default is `{{.Tag}}`
+  name_template: "Meshery {{.Tag}}"
```
Comment (leecalcote) [inline .goreleaser.yml]: Looks good.

File: Makefile [hunk 2]
```diff
@@ -73,3 +78,7 @@ run-ui-dev:
 # Builds the user interface on your local machine.
 build-ui:
 	cd ui; npm run build && npm run export; cd ..
+
+# setup wrk for local dev
+setup-wrk:
```
Comment (leecalcote) [inline Makefile]: Is this setup for wrk or wrk2?

Anything needed for building wrk2?

File: helpers/fortio.go [hunk 3]
```diff
@@ -30,10 +30,10 @@ func sharedHTTPOptions(opts *models.LoadTestOptions) *fhttp.HTTPOptions {
 	httpOpts.ContentType = ""
 	// httpOpts.Payload = fnet.GeneratePayload(*PayloadFileFlag, *PayloadSizeFlag, *PayloadFlag)
 	// httpOpts.UnixDomainSocket = *unixDomainSocketFlag
-	if false { // *followRedirectsFlag {
-		httpOpts.FollowRedirects = true
-		httpOpts.DisableFastClient = true
-	}
+	// if false { // *followRedirectsFlag {
+	httpOpts.FollowRedirects = true
```
Comment (leecalcote) [inline helpers/fortio.go]: @girishranganathan quick check - have you verified that the desired redirect behavior happens with an integration test?

**Comment by leecalcote** [comment]
**Output from wrk2 perf test when running `make run-local`:**

```
ERRO[0041] Unable to read data from bitcask store: error: key not found
WARN[0041] Unable to read session from the session persister. Starting a new session.
DEBU[0041] received command: wrk [-t1 -d30s -R-1 -s ./wrk2/scripts/multiple-endpoints_in_json.lua https://www.google.com:443]
DEBU[0041] received new data on response channel         file=load_test_handler
DEBU[0041] Flushed the messages on the wire...           file=load_test_handler
ERRO[0041] unable to execute the requested command: fork/exec ./wrk2/wrk: no such file or directory
ERRO[0041] error while running tests: unable to execute the requested command: fork/exec ./wrk2/wrk: no such file or directory
ERRO[0041] error: unable to perform load test: error while running tests: unable to execute the requested command: fork/exec ./wrk2/wrk: no such file or directory
DEBU[0041] received new data on response channel         file=load_test_handler
DEBU[0041] Flushed the messages on the wire...           file=load_test_handler
DEBU[0041] response channel closed                       file=load_test_handler
DEBU[0041] load test completed                           file=load_test_handler
```

**Comment by leecalcote** [comment]
Icon available - https://github.com/layer5io/meshery/pull/292

File: Makefile (hunk 2 above)
Comment (girishranganathan) [reply Makefile]: I started off with this, but it does not work on a Mac atm. I will get this fixed one way or the other.

File: helpers/fortio.go (hunk 3 above)
Comment (girishranganathan) [reply helpers/fortio.go]: Yes

**Comment by girishranganathan** [comment]
> **Output from wrk2 perf test when running `make run-local`:**
> 
> ```
> ERRO[0041] Unable to read data from bitcask store: error: key not found
> WARN[0041] Unable to read session from the session persister. Starting a new session.
> DEBU[0041] received command: wrk [-t1 -d30s -R-1 -s ./wrk2/scripts/multiple-endpoints_in_json.lua https://www.google.com:443]
> DEBU[0041] received new data on response channel         file=load_test_handler
> DEBU[0041] Flushed the messages on the wire...           file=load_test_handler
> ERRO[0041] unable to execute the requested command: fork/exec ./wrk2/wrk: no such file or directory
> ERRO[0041] error while running tests: unable to execute the requested command: fork/exec ./wrk2/wrk: no such file or directory
> ERRO[0041] error: unable to perform load test: error while running tests: unable to execute the requested command: fork/exec ./wrk2/wrk: no such file or directory
> DEBU[0041] received new data on response channel         file=load_test_handler
> DEBU[0041] Flushed the messages on the wire...           file=load_test_handler
> DEBU[0041] response channel closed                       file=load_test_handler
> DEBU[0041] load test completed                           file=load_test_handler
> ```

Yeah you cannot test it on a Mac. Try running it from a Docker container.

**Comment by leecalcote** [comment]
Running `make docker` produces this error:

```
 => CACHED [bd0 2/5] RUN adduser --disabled-login appuser                                                                     0.0s
 => CACHED [bd0 3/5] WORKDIR /github.com/layer5io/meshery                                                                     0.0s
 => CACHED [bd0 4/5] ADD . .                                                                                                  0.0s
 => ERROR [bd0 5/5] RUN cd cmd; GOPROXY=direct GOSUMDB=off go build -ldflags="-w -s" -tags draft -a -o /meshery .           186.5s
 => CACHED [bd1 2/3] ADD ui ui                                                                                                0.0s
 => CACHED [bd1 3/3] RUN cd ui; npm i; npm run build && npm run export; mv out /                                              0.0s
------                                                                                                                             
 > [bd0 5/5] RUN cd cmd; GOPROXY=direct GOSUMDB=off go build -ldflags="-w -s" -tags draft -a -o /meshery .:                        
#13 186.3 go: golang.org/x/exp@v0.0.0-20191030013958-a1ab85dbe136 requires
#13 186.3       golang.org/x/mod@v0.1.0: reading golang.org/x/mod/go.mod at revision v0.1.0: unknown revision v0.1.0
------
failed to solve with frontend dockerfile.v0: failed to build LLB: executor failed running [/bin/sh -c cd cmd; GOPROXY=direct GOSUMDB=off go build -ldflags="-w -s" -tags draft -a -o /meshery .]: runc did not terminate sucessfully
```

**Pull request merged**